Assets/Scripts/Enemies/ChomperEnemy.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+/// <summary>
+/// Enemy Chomper is created and checking for player collision with Enemy and updating player lives
+/// </summary>
+public class ChomperEnemy : Enemy
+{
+    // Initializing target pos
+    private Vector3 _currentTarget;
+    
+    private Animator _animator;
+    
+    private SpriteRenderer _chomperSprite;
+    // Intializing the playercontroller
+    private PlayerController _playercontroller;
+    // Start is called before the first frame update
+    void Start()
+    {
+        // getting the Animator component
+        _animator = GetComponent<Animator>();
+        // getting the SpriteRender
+        _chomperSprite = GetComponent<SpriteRenderer>();
+    }
+
+    // Update is called once per frame
+    public override void Update()
+    {
+       // Debug.Log("ChomperEnemey updated......");
+       // Getting the state of the Animator and checking for Idle state
+        if(_animator.GetCurrentAnimatorStateInfo(0).IsName("ChomperIdle_anim"))
+        {
+            // do nothing
+            return;
+        }
+       // Creating the Patrol AI for Enemy
+        movementEnemy();
+    }
+    private void movementEnemy()
+    {
+        flipChomper();
+        // checking if th position is in point A
+        if (transform.position == pointA.position)
```

Comment:
We should not compare the postions here try to do this in different way with time or with raycast.